stats/lib/platform-stats-fetchers.js

```diff
@@ -77,6 +77,40 @@ export const fetchDailyRewardTransfers = async (pgPool, filter) => {
   return rows
 }
 
+/**
+ * @param {Queryable} pgPool
+ * @param {import('./typings.js').DateRangeFilter} filter
+ */
+export const fetchAccumulativeDailyParticipantCount = async (pgPool, filter) => {
+  const { rows } = await pgPool.query(`
+    WITH daily_participants_count AS (
+      SELECT
+        day,
+        participant_id,
+        ROW_NUMBER() OVER (PARTITION BY participant_id ORDER BY day) AS rn
+      FROM daily_participants
+    ),
+    first_appearance AS (
+      SELECT day, participant_id
+      FROM daily_participants_count
+      WHERE rn = 1
+    ),
+    cumulative_participants AS (
+      SELECT
+        day,
+        COUNT(participant_id) OVER (ORDER BY day) AS cumulative_participants
```

Comment:
Nice trick! It took me a bit to decipher this. The PG documentation was helpful.

https://www.postgresql.org/docs/current/tutorial-window.html
> There is another important concept associated with window functions: for each row, there is a set of rows within its partition called its window frame. Some window functions act only on the rows of the window frame, rather than of the whole partition. By default, if ORDER BY is supplied then the frame consists of all rows from the start of the partition up through the current row, plus any following rows that are equal to the current row according to the ORDER BY clause.

Having said that, I think there may be a subtle bug there. IIUC, this counts all `(day, participant_id)` pairs - i.e. if a participant was active on two days, they will contribute value `2` to the final sum.

https://www.postgresql.org/docs/current/functions-aggregate.html
> `count ( "any" ) → bigint`
> Computes the number of input rows in which the input value is not null.

Can you please check?

I would expect to see `COUNT(DISTINCT participant_id)` in the query instead. That can be slow, this StackOverflow answer suggest a faster option: https://stackoverflow.com/a/14732410/69868

Or maybe the current behaviour is what you actually want. Can you please clarify?

Reply:
Can you verify this by looking at the tests? https://github.com/filecoin-station/spark-stats/pull/267/files#diff-bcf599781f16d5a58a3aea9fc84d1f3633de99d563fa617c59023ee1ddde4f0b Do you agree that this count is what's expected? I think so.

Tbh I haven't looked at the query in the depth that you did for the review.

Reply:
Would you still like me to compare the current version with what's proposed, even if the output is correct? Would you like to simplify or speed up the query?